src/laserscan_multi_merger.cpp

```diff
@@ -208,21 +210,34 @@ void LaserscanMerger::pointcloud_to_laserscan(Eigen::MatrixXf points, pcl::PCLPo
 			ROS_DEBUG("rejected for angle %f not in range (%f, %f)\n", angle, output->angle_min, output->angle_max);
 			continue;
 		}
-		int index = (angle - output->angle_min) / output->angle_increment;
 
+		int index = (angle - output->angle_min) / output->angle_increment;
 
 		if (output->ranges[index] * output->ranges[index] > range_sq)
+		{
 			output->ranges[index] = sqrt(range_sq);
+			intensities.resize(ranges_size, 0.0f);
+		}
+
+		if (intensities[index] < intensity_value)
+		{
+			intensities[index] = intensity_value;
+		}
 	}
 
-	laser_scan_publisher_.publish(output);
+        output->intensities.resize(intensities.size());
+        for (size_t i = 0; i < intensities.size(); ++i)
+        {
+            output->intensities[i] = intensities[i];
+        }
```

Comment:
does this not work?
`output->intensities = intensities`

Reply:
Thanks, yeah, direct assignment works too. I have tested and updated it.